http.go

```diff
@@ -315,20 +315,26 @@ func (resp *Response) LocalAddr() net.Addr {
 	return resp.laddr
 }
 
-// Body returns response body.
-//
-// The returned body is valid until the response modification.
-func (resp *Response) Body() []byte {
+// BodyOrErr returns response (body, nil) if success.
+// when error occurs, it returns nil, err.
+func (resp *Response) BodyOrErr() ([]byte, error) {
 	if resp.bodyStream != nil {
 		bodyBuf := resp.bodyBuffer()
 		bodyBuf.Reset()
 		_, err := copyZeroAlloc(bodyBuf, resp.bodyStream)
 		resp.closeBodyStream() //nolint:errcheck
 		if err != nil {
-			bodyBuf.SetString(err.Error())
+			return nil, err
 		}
 	}
-	return resp.bodyBytes()
+	return resp.bodyBytes(), nil
+}
+
+// Body returns response body.
+// if get body failed, returns nil.
+func (resp *Response) Body() []byte {
+	body, _ := resp.BodyOrErr()
```

Comment:
Now this function isn't backwards compatible anymore as it doesn't do the `bodyBuf.SetString(err.Error())` anymore. Can you change it back to how it was? I don't think it's easy to use `BodyOrErr` for that so I would just leave the function as it was.

Reply:
i recommit again,  the Body() function returns body with err msg when error happens.